seatunnel-e2e/seatunnel-flink-sql-e2e/setunnel-connector-flink-sql-e2e-base/src/test/java/org/apache/seatunnel/e2e/flink/sql/FlinkContainer.java

```diff
@@ -17,128 +17,49 @@
 
-        Startables.deepStart(Stream.of(jobManager)).join();
-        Startables.deepStart(Stream.of(taskManager)).join();
-        copySeaTunnelFlinkFile();
-        LOG.info("Flink containers are started.");
-    }
-
-    @AfterEach
-    public void close() {
-        if (taskManager != null) {
-            taskManager.stop();
-        }
-        if (jobManager != null) {
-            jobManager.stop();
-        }
-    }
+    private static final String CONNECTORS_ROOT_PATH = "seatunnel-connectors/seatunnel-connectors-flink-sql";
 
-    public Container.ExecResult executeSeaTunnelFlinkSqlJob(String confFile)
-        throws IOException, InterruptedException, URISyntaxException {
-        final String confPath = Paths.get(FlinkContainer.class.getResource(confFile).toURI()).toString();
-        if (!new File(confPath).exists()) {
-            throw new IllegalArgumentException(confFile + " doesn't exist");
-        }
-        final String targetConfInContainer = Paths.get("/tmp", confFile).toString();
-        jobManager.copyFileToContainer(MountableFile.forHostPath(confPath), targetConfInContainer);
+    private static final String CONNECTOR_TYPE = "seatunnel-sql";
 
-        // Running IT use cases under Windows requires replacing \ with /
-        String conf = targetConfInContainer.replaceAll("\\\\", "/");
-        final List<String> command = new ArrayList<>();
-        command.add(Paths.get(SEATUNNEL_HOME, "bin/start-seatunnel-sql.sh").toString());
-        command.add("--config " + conf);
+    private static final String CONNECTOR_PREFIX = "flink-sql-connector-";
 
-        Container.ExecResult execResult = jobManager.execInContainer("bash", "-c", String.join(" ", command));
-        LOG.info(execResult.getStdout());
-        LOG.error(execResult.getStderr());
-        // wait job start
-        Thread.sleep(WAIT_FLINK_JOB_SUBMIT);
-        return execResult;
+    public FlinkContainer() {
+        super(FLINK_DOCKER_IMAGE,
+            START_SHELL_NAME,
+            START_MODULE_NAME,
+            CONNECTORS_ROOT_PATH,
+            CONNECTOR_TYPE,
+            CONNECTOR_PREFIX);
     }
 
-    protected void copySeaTunnelFlinkFile() {
-        // copy lib
-        String seatunnelCoreFlinkJarPath = PROJECT_ROOT_PATH + "/seatunnel-core/seatunnel-core-flink-sql/target/" + SEATUNNEL_FLINK_SQL_JAR;
-        jobManager.copyFileToContainer(
-            MountableFile.forHostPath(seatunnelCoreFlinkJarPath),
-            FLINK_JAR_PATH);
-
-        // copy bin
-        String seatunnelFlinkBinPath = PROJECT_ROOT_PATH + "/seatunnel-core/seatunnel-core-flink-sql/src/main/bin/start-seatunnel-sql.sh";
-        jobManager.copyFileToContainer(
-            MountableFile.forHostPath(seatunnelFlinkBinPath),
-            Paths.get(SEATUNNEL_BIN).toString());
+    @Override
+    public Container.ExecResult executeSeaTunnelFlinkJob(String confFile) throws IOException, InterruptedException {
```

Comment:
No need to call `copyConnectorJarToContainer`?

Reply:
The old `flink-sql-starter` don’t call copyConnectorJarToContainer.

Reply:
👌

Reply:
Without this logic is really wrong, I've added `copyConnectorJarToContainer`.
There are other problems with flink-sql-e2e. At present, it only uses the built-in connector of flink, so there is no error. I will solve this problem in other PRs.